sdk/loadtestservice/load-testing-rest/src/beginTestRunStatus.ts

```diff
@@ -0,0 +1,175 @@
+): Promise<TestRunStatusPoller> {
+  let getTestRunResult = await client.path("/test-runs/{testRunId}", testRunId).get();
+  if (isUnexpected(getTestRunResult)) {
+    throw getTestRunResult.body.error;
+  }
+  //let TestRunStatusResponse: TestRunGet200Response;
+
+  type Handler = (state: OperationState<TestRunGet200Response>) => void;
+
+  const state: OperationState<TestRunGet200Response> = {
+    status: "notStarted",
+  };
+
+  const progressCallbacks = new Map<symbol, Handler>();
+  const processProgressCallbacks = async (): Promise<void> =>
+    progressCallbacks.forEach((h) => h(state));
+  let resultPromise: Promise<TestRunGet200Response> | undefined;
+  let cancelJob: (() => void) | undefined;
+  const abortController = new AbortController();
+  const currentPollIntervalInMs = polledOperationOptions.updateIntervalInMs ?? 2000;
+
+  const poller: SimplePollerLike<OperationState<TestRunGet200Response>, TestRunGet200Response> = {
+    async poll(_options?: { abortSignal?: AbortSignalLike }): Promise<void> {
+      getTestRunResult = await client.path("/test-runs/{testRunId}", testRunId).get();
+      if (isUnexpected(getTestRunResult)) {
+        throw getTestRunResult.body.error;
+      }
+
+      if (getTestRunResult.body.status === "FAILED") {
+        state.status = "failed";
+        state.error = new Error(getTestRunResult.body.status);
+      }
+
+      if (
+        getTestRunResult.body.status === "CANCELLING" ||
+        getTestRunResult.body.status === "CANCELLED"
+      ) {
+        state.status === "canceled";
+      }
+
+      if (getTestRunResult.body.status === "DONE") {
+        state.status = "succeeded";
+        state.result = getTestRunResult;
+      }
+
+      await processProgressCallbacks();
+
+      if (state.status === "canceled") {
+        throw new Error("Operation was canceled");
+      }
+      if (state.status === "failed") {
+        throw state.error;
+      }
+    },
+
+    pollUntilDone(pollOptions?: { abortSignal?: AbortSignalLike }): Promise<TestRunGet200Response> {
+      return (resultPromise ??= (async () => {
+        const { abortSignal: inputAbortSignal } = pollOptions || {};
+        const { signal: abortSignal } = inputAbortSignal
+          ? new AbortController([inputAbortSignal, abortController.signal])
+          : abortController;
+        if (!poller.isDone()) {
+          await poller.poll({ abortSignal });
+          while (!poller.isDone()) {
+            const delay = sleep(currentPollIntervalInMs, abortSignal);
+            cancelJob = () => abortController.abort();
+            await delay;
+            await poller.poll({ abortSignal });
+          }
+        }
+        switch (state.status) {
+          case "succeeded": {
+            return poller.getResult() as TestRunGet200Response;
+          }
+          case "canceled": {
+            throw new Error("Operation was canceled");
+          }
+          case "failed": {
+            throw state.error;
+          }
```

Comment:
Same thing applies here too:
```suggestion
          case "succeeded":
          case "failed":
          case "canceled": {
            return poller.getResult() as TestRunGet200Response;
          }
```